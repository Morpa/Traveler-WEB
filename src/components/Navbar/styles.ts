import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.nav`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: flex-start;
    background: ${theme.colors.primary};
  `}

  ${media.lessThan('medium')`
    justify-content: center;
  `}
`
