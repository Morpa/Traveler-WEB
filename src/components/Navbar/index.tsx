import Link from 'next/link'

import Logo from 'components/Logo'

import * as S from './styles'

export type NavbarProps = {
  children?: React.ReactNode
}

const Navbar = ({ children }: NavbarProps) => (
  <S.Wrapper>
    <Link href="/" passHref>
      <a>
        <Logo />
      </a>
    </Link>

    {!!children && <>{children}</>}
  </S.Wrapper>
)

export default Navbar
