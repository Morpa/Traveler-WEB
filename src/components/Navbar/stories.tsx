import { Story, Meta } from '@storybook/react/types-6-0'
import Navbar, { NavbarProps } from '.'

export default {
  title: 'Navbar',
  component: Navbar
} as Meta

export const Default: Story<NavbarProps> = () => <Navbar />
