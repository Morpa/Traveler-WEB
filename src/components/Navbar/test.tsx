import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Navbar from '.'

describe('<Navbar />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <Navbar>
        <div>my content</div>
      </Navbar>
    )

    expect(screen.getByLabelText(/traveler/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
