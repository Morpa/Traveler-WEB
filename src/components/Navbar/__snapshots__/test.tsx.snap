// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render the heading 1`] = `
.c1 {
  width: 20rem;
  height: 5.9rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background: #FFFFFF;
}

@media (max-width:768px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<nav
  class="c0"
>
  <a
    href="/"
  >
    <div
      class="c1"
    >
      <svg
        aria-label="Traveler"
        fill="none"
        role="img"
        viewBox="0 0 126 26"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M10.2889 11.3051H6.75974c-1.35737 6.519-2.06593 9.898-2.06593 10.1642 0 1.2304 1.49312 1.1435 2.90208.3069l.92302 3.1237c-.87958.747-2.0225 1.0865-3.3853 1.0865-2.77177 0-4.178016-1.4966-4.178016-4.1803.010859-.6465.673256-4.1206 2.030636-10.501H0l.746558-3.60723H3.78166l.74655-3.69411L5.4078 0h3.78165c-.13393.747876-.39816 2.02361-.79271 3.8272L7.5606 7.69787h3.4749l-.7466 3.60723zM25.7201 7.60554l-1.6289 3.30026c-.5295-.1483-1.0799-.208-1.6288-.1766-1.6093-.0073-3.1952.3857-4.6151 1.1436l-2.8098 13.4237h-3.7816l3.7355-17.59861h3.1219l-.0434.61659c1.1585-.63657 2.4354-1.028 3.7515-1.14996 1.316-.12196 2.6431.02816 3.8987.44102zM67.7438 17.289c-.0871.5525-.1316 1.111-.133 1.6705 0 2.2907 1.0551 3.4351 3.1654 3.4333 2.5953 0 4.2648-1.3201 4.6612-3.2133l3.6921.3531C78.2064 23.8433 75.3505 26 70.6459 26c-4.3979 0-6.8168-2.8168-6.8168-7.084 0-3.2595.9674-6.031 2.9021-8.3145 1.9356-2.33052 4.4413-3.47406 7.4737-3.47406 4.178 0 6.1516 2.65107 6.1516 6.37776-.0213 1.2727-.1686 2.5401-.4398 3.7838H67.7438zm1.0099-3.1645h7.8728c.0862-.4054.1308-.8186.133-1.2332 0-1.4504-.923-2.154-2.8152-2.154-2.332-.0027-4.0477 1.1409-5.1906 3.3872zM86.7795 21.4584c0-.1983 1.5039-7.3773 4.5065-21.4584h-3.7736c-.704 3.30659-1.4759 6.8848-2.3157 10.7346-1.4496 6.7309-2.1528 10.425-2.1528 11.0851 0 2.6836 1.4063 4.1803 4.1781 4.1803 1.3573 0 2.503-.3531 3.3826-1.0865l-.9231-3.1237c-1.4062.8203-2.902.9099-2.902-.3314zM97.1588 17.289c-.0871.5525-.1315 1.111-.133 1.6705 0 2.2907 1.0551 3.4351 3.1652 3.4333 2.596 0 4.265-1.3201 4.661-3.2133l3.692.3531C107.621 23.8433 104.766 26 100.061 26c-4.398 0-6.8169-2.8168-6.8169-7.084 0-3.2595.9674-6.031 2.9021-8.3145 1.9356-2.33052 4.4418-3.47406 7.4768-3.47406 4.175 0 6.154 2.63749 6.154 6.37776-.021 1.2727-.169 2.5401-.44 3.7838H97.1588zm1.0099-3.1645h7.8723c.087-.4054.131-.8186.134-1.2332 0-1.4504-.923-2.154-2.816-2.154-2.332-.0027-4.0474 1.1409-5.1903 3.3872zM126 7.60556l-1.628 3.30024c-.53-.1483-1.08-.2079-1.629-.1765-1.61-.0074-3.196.3856-4.615 1.1435l-2.813 13.4183h-3.779l3.736-17.59862h3.122l-.03.62202c1.156-.63469 2.431-1.02508 3.745-1.14703 1.313-.12194 2.638.02716 3.891.43809z"
          fill="#115D8C"
        />
        <path
          d="M22.9886 25.9619l-1.7972-2.6537L34.0322 5.19073c.5372-.75931 1.2819-1.34744 2.1448-1.69385.8629-.3464 1.8073-.43638 2.7201-.25914l4.9354 20.60276-2.0008 2.078-10.1966-2.0399-8.6465 2.0833zm8.9587-5.5167l7.9406 1.589-3.4423-14.32822-9.8192 13.85292 4.5635-1.0865.7574-.0272zM47.8612 25.8751L42.9258 5.27227l2.0008-2.07793 10.1966 2.03991 8.6438-2.08337 1.7972 2.65378L52.7261 23.9221c-.5379.7585-1.2827 1.3461-2.1454 1.6924-.8627.3463-1.8068.4368-2.7195.2606zm-.9827-18.79652l3.4314 14.32822 9.822-13.85287-4.5743 1.1028-.7385.0163-7.9406-1.59445z"
          fill="#F25D27"
        />
      </svg>
    </div>
  </a>
  <div>
    my content
  </div>
</nav>
`
